From f2883d73cd68639cd82a668600ad684f61655013 Mon Sep 17 00:00:00 2001
From: Jianzheng Zhou <B38613@freescale.com>
Date: Thu, 19 Jul 2012 17:00:56 +0800
Subject: [PATCH 312/502] ENGR00217130 GPS:display Localtime error

For some timezone,GPS show localtime incorrect.This is caused
by transfering UTC to Localtime.UTC is got from GPS parser.
then mktime it to get localtime. At last this localtime is
reported to framwork. add tm_isdst = -1 means mktime should
use timezone data to determine whether DST should be considered
at the specified time.

Signed-off-by: Jianzheng Zhou <B38613@freescale.com>
---
 libgps/athr_gps.c | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/libgps/athr_gps.c b/libgps/athr_gps.c
index 50b799b..b46a006 100755
--- a/libgps/athr_gps.c
+++ b/libgps/athr_gps.c
@@ -402,6 +402,7 @@ static int nmea_reader_get_timestamp(NmeaReader*  r, Token  tok, time_t *timesta
     tm.tm_year = r->utc_year - 1900;
     tm.tm_mon  = r->utc_mon - 1;
     tm.tm_mday = r->utc_day;
+    tm.tm_isdst = -1;
 
     // D("h: %d, m: %d, s: %f", tm.tm_hour, tm.tm_min, tm.tm_sec);
     // D("Y: %d, M: %d, D: %d", tm.tm_year, tm.tm_mon, tm.tm_mday);
@@ -451,7 +452,6 @@ nmea_reader_update_date( NmeaReader*  r, Token  date, Token  mtime )
     int    day, mon, year;
 
     if (tok.p + 6 != tok.end) {
- 
         D("no date info, use host time as default: '%.*s'", tok.end-tok.p, tok.p);
         /* no date info, will use host time in _update_time function
            jhung 2010/05/12
@@ -792,7 +792,6 @@ nmea_reader_parse( NmeaReader*  r )
           int totalSentences = str2int(tok_noSentences.p, tok_noSentences.end);
           int curr;
           int i;
- 
           if (sentence == 1) {
               r->sv_status_changed = 0;
               r->sv_status.num_svs = 0;
@@ -1717,7 +1716,6 @@ static int athr_gps_xtra_inject_data(char *data, int length)
     int cnt;
 
     D("%s: entered, data: %.2x (len: %d)", __FUNCTION__, *data, length);
- 
     if (write(s->fd, data, length) < 0)
     {
       D("Send $PUNV command ERROR!");
-- 
1.8.0

